### PR TITLE
[lldb][test] Modernize asserts

### PR DESCRIPTION
**Pull request by rupprecht**
This uses [teyit](https://pypi.org/project/teyit/) to modernize asserts, as recommended by the [unittest release notes](https://docs.python.org/3.12/whatsnew/3.12.html#id3).

For example, `assertTrue(a == b)` is replaced with `assertEqual(a, b)`. This produces better error messages, e.g. `error: unexpectedly found 1 and 2 to be different` instead of `error: False`.

### CODE REVIEW

File: lldb/test/API/macosx/queues/TestQueues.py [hunk 1]
```diff
@@ -456,8 +457,9 @@ def queues_with_libBacktraceRecording(self):
             "doing_the_work_2",
             "queue 2's pending item #0 should be doing_the_work_2",
         )
-        self.assertTrue(
-            queue_performer_2.GetPendingItemAtIndex(9999).IsValid() == False,
+        self.assertEqual(
+            queue_performer_2.GetPendingItemAtIndex(9999).IsValid(),
```
Comment (bulbazord) [inline lldb/test/API/macosx/queues/TestQueues.py:461]: Might be useful to do an additional pass over these later. I think we could use `assertFalse` here, for example.

**Review by bulbazord — APPROVED** [review]
I did a quick manual inspection of all of them, the transformation produced by Teyit looks correct to me. I did notice a few places where we were doing convoluted checks (I left a comment on one) but those should be addressed in follow-ups I think.

File: lldb/test/API/macosx/queues/TestQueues.py (hunk 1 above)
Comment (rupprecht) [reply lldb/test/API/macosx/queues/TestQueues.py:461]: Good point -- I can check for this kind of case as a followup.

Comment (rupprecht) [reply lldb/test/API/macosx/queues/TestQueues.py:461]: https://github.com/llvm/llvm-project/pull/82526 includes this and more

**Pull request merged**